### DISCUSSION TITLE
Issue with APNs Push Notification

### DISCUSSION THREAD

**Discussion by pyjun01** [magicbell/community#161] [Questions] [answered]
Hi team,

I have an issue with APNs Push Notification.

When I attempt to send a push notification to user's iOS device using APNs token on the Compose page, the user push subscription gets automatically discarded. Additionally, there's no "mobile_push" log on the Logs page, even though the notification delivery status is "sent".

Here are some screenshots for reference:
<img src="https://github.com/magicbell-io/community/assets/32596517/2f7f902c-a6b0-403a-8b90-1d6942533581" width="500" />
<img src="https://github.com/magicbell-io/community/assets/32596517/f61b6600-0498-4c6f-85c4-7fcd859e5dd0" width="500" />
<img src="https://github.com/magicbell-io/community/assets/32596517/0a5ec10d-2744-44e0-a6a7-a5e9cfc5489a" width="500" />

If I send notification directly to iOS device via CloudKit Console, the notification appear in my app. (APNs Token is valid)

Is there something wrong with my APNs provider settings?
- p8 cert format: "-----BEGIN PRIVATE KEY----- ***** -----END PRIVATE KEY-----" (this format is based on [previous discussion](https://github.com/orgs/magicbell-io/discussions/129))
- environment: I've tried all options(blank, "development", "production")

**Comment by unamashana** [comment]
Hey @pyjun01 sorry about that! We found an issue and pushed up a fix. Can you please try again? This was an application logic error which was also eating up the log.

**Comment by reykeem** [comment]
Hello,

I am experiencing the same thing, with the device token in user push notifications getting discarded as soon as we attempt to send a notification via MagicBell console. Is there a fix or logs to debug?

**Reply by unamashana** [reply]
Hey @reykeem we deployed a fix around the time you posted. Can you please confirm if you are still seeing this issue?

**Reply by reykeem** [reply]
Hey @unamashana still seeing issues:

What I have tried:

Configurations per magicbell docs for react native
APN settings added to delivery channel
Env set to prod/dev/blank
Notis sent from CloudKit show up on sim/device
Notis sent from MagicBell console dosent work and discards the device token from the push subscriptions list after attempt to send noti
No error logs

**Reply by unamashana** [reply]
Sorry about that - we will get back to you this week with a resolution.

**Reply by reykeem** [reply]
@unamashana  Hey Hana, any update on this issue?

**Reply by unamashana** [reply]
Hey @reykeem applpgies for the silence. The logging in our current set of integrations can be a bit patchy and make debugging some issues harder. We've been working on v2 of our most used providers and iOS is one of them. Would you be open to beta testing the new version? We can help you on magicbell.to/slack.

**Reply by unamashana** [reply, marked as answer]
For anyone else browsing this issue, the new APNS integration is currently open to testing and feedback. We are expecting to take it to production within 2 weeks. We've give @reykeem access to it on our Slack and happy to do so with anyone else - https://magicbell.to/slack.
